transaction-service/src/main/java/pl/adamboguszewski/transaction/service/application/TransactionRepository.java

```diff
@@ -0,0 +1,15 @@
+package pl.adamboguszewski.transaction.service.application;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+public interface TransactionRepository<Transaction> {
```

Comment:
Big changes in repository, look for TransactionRepository<Transaction> and TransactionJpaRepository* classes implementation. Also your error was generated by I-don't-fucking-know-why-but-my-implementation-is-better-anyway